modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceDetailsViewManager.java

```diff
@@ -175,7 +191,20 @@ public VendorHostedDeployedServiceDetails getVendorHostedServiceDetailsByIdForEn
             log.error(errorMsg);
             throw new ServiceDetailsNotAccessible(errorMsg);
         }
-        return EntityTransUtils.transToVendorHostedServiceDetails(serviceDeploymentEntity);
+        VendorHostedDeployedServiceDetails details =
+                EntityTransUtils.transToVendorHostedServiceDetails(serviceDeploymentEntity);
+        ServiceTemplateEntity serviceTemplateEntity = serviceTemplateStorage
```

Comment:
It is better to add a method to query all templates in batches, instead of invoking the query SQL statement one by one in a loop. Or we can modify the serviceTemplateId  to serviceTemplateEntity with @oneToone in the ServiceDeploymentEntity, in this way, ServiceTemplateEntity will be as a part of ServiceDeploymentEntity.